fe/fe-core/src/main/java/org/apache/doris/planner/external/hudi/HudiScanNode.java

```diff
@@ -0,0 +1,231 @@
+        computeColumnFilter();
+        initBackendPolicy();
+        if (table instanceof HMSExternalTable) {
+            source = new HudiHMSSource((HMSExternalTable) table, desc, columnNameToRange);
+        }
+        Preconditions.checkNotNull(source);
+        initSchemaParams();
+    }
+
+
+    public static void setHudiParams(TFileRangeDesc rangeDesc, HudiSplit hudiSplit) {
+        TTableFormatFileDesc tableFormatFileDesc = new TTableFormatFileDesc();
+        tableFormatFileDesc.setTableFormatType(hudiSplit.getTableFormatType().value());
+        THudiFileDesc fileDesc = new THudiFileDesc();
+        fileDesc.setBasePath(hudiSplit.getBasePath());
+        fileDesc.setDataFilePath(hudiSplit.getDataFilePath());
+        fileDesc.setDeltaFilePaths(hudiSplit.getHudiDeltaLogs());
+        tableFormatFileDesc.setHudiParams(fileDesc);
+        rangeDesc.setTableFormatParams(tableFormatFileDesc);
+    }
+
+    @Override
+    public List<Split> getSplits() throws UserException {
+        // TODO expressions
+        // List<Expression> expressions = new ArrayList<>();
+
+        // for (Expr conjunct : conjuncts) {
+        //     Expression expression = null;
+        //     if (expression != null) {
+        //         expressions.add(expression);
+        //     }
+        // }
+
+        HMSExternalTable targetTable = (HMSExternalTable) source.getTargetTable();
+        HoodieTableMetaClient hudiClient = HiveMetaStoreClientHelper.getHudiClient(targetTable);
+
+        hudiClient.reloadActiveTimeline();
+
+        String partitionName = getHudiPartitionName(targetTable, hudiClient);
+
+        List<Split> splits = new ArrayList<>();
+
+        String basePath = targetTable.getRemoteTable().getSd().getLocation();
+        String inputFormat = targetTable.getRemoteTable().getSd().getInputFormat();
+        String serdeLib = targetTable.getRemoteTable().getSd().getSerdeInfo().getSerializationLib();
+
+        TableSchemaResolver schemaUtil = new TableSchemaResolver(hudiClient);
+        Schema hudiSchema;
+        try {
+            hudiSchema = HoodieAvroUtils.createHoodieWriteSchema(schemaUtil.getTableAvroSchema());
+        } catch (Exception e) {
+            throw new RuntimeException("Cannot get hudi table schema.");
+        }
+
+        List<String> columnNames = new ArrayList<>();
+        List<String> columnTypes = new ArrayList<>();
+        List<FieldSchema> allFields = targetTable.getRemoteTable().getSd().getCols();
+        allFields.addAll(targetTable.getRemoteTable().getPartitionKeys());
+
+        for (Schema.Field hudiField : hudiSchema.getFields()) {
+
+            String columnName = hudiField.name().toLowerCase(Locale.ROOT);
+            Optional<FieldSchema> field = allFields.stream()
+                    .filter(f -> f.getName().equals(columnName)).findFirst();
+            if (!field.isPresent()) {
+                String errorMsg = String.format("Hudi column %s not exists in hive metastore.", hudiField.name());
```

Comment:
How is this situation caused? Can the [synchronization tools](https://hudi.apache.org/docs/syncing_metastore#hive-sync-tool) maintain strong consistency in the table schema of the hive schema?